golos.publication/golos.publication.cpp

```diff
@@ -412,6 +412,15 @@ void publication::set_vote(account_name voter, account_name author, string perml
                 item.state.netshares = new_mssg_rshares.data();
                 item.state.sumcuratorsw = (FP(item.state.sumcuratorsw) - FP(vote_itr->curatorsw)).data();
             });
+
+            if (is_account(config::social_name)) {
+                action(
+                    permission_level{_self, N(active)},
+                    config::social_name,
+                    N(setreput),
+                    std::make_tuple(voter, author, vote_itr->rshares, rshares.data())
```

Comment:
You should get calculation from here:
https://github.com/GolosChain/golos/blob/master/plugins/follow/plugin.cpp#L241-L283

Reply:
@afalaleev If rules aren't conforming - we should just don't add reputation as in follow, or throw exception with assert?

Reply:
It is necessary to transfer division rshares by 128 (`>>6`) into this code.